lib/vm/src/memory.rs

```diff
@@ -155,7 +177,11 @@ struct WasmMmap {
 
 impl LinearMemory {
     /// Create a new linear memory instance with specified minimum and maximum number of wasm pages.
-    pub fn new(memory: &MemoryType, style: &MemoryStyle) -> Result<Self, MemoryError> {
+    pub fn new(
+        memory: &MemoryType,
+        style: &MemoryStyle,
+        vm_memory_location: Option<NonNull<VMMemoryDefinition>>,
+    ) -> Result<Self, MemoryError> {
```

Comment:
I don't love this API. I think we should divide into two functions.

```rust
    pub fn new(memory: &MemoryType, style: &MemoryStyle) -> Result<Self, MemoryError> {
```

And:

```rust
    pub fn from_definition(memory: &MemoryType, style: &MemoryStyle, definition: `VMMemoryDefinition`) -> Result<Self, MemoryError> {
```

(same for tables)

Reply:
Hmm, yeah that makes sense -- I'm open to that too